packages/core/src/builder.class.ts

```diff
@@ -984,6 +994,20 @@ export class Builder {
 
   static registerAction(action: Action) {
     this.actions.push(action);
+
+    if (Builder.isBrowser) {
+      const actionClone = JSON.parse(JSON.stringify(action));
```

Comment:
Can you try this instead?
```suggestion
      const actionClone = {
         ...action,
         action: action.action?.toString(),
    };
```

Reason: It's a performance hit for large objects

Reply:
@yash-builder This is shallow clone. It will cause issue for the postMessage.